packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/CustomVis/components/CustomVisTemplate.tsx

```diff
@@ -0,0 +1,143 @@
+        case TemplateType.BAR_CHART:
+            return IconChartBar;
+        case TemplateType.HEATMAP:
+            return IconFlame;
+        case TemplateType.BUBBLE_PLOTS:
+            return IconChartBubble;
+        case TemplateType.FUNNEL_CHART:
+            return IconFilter;
+        case TemplateType.WATERFALL_CHART:
+            return IconStairs;
+        case TemplateType.MAP:
+            return IconWorld;
+    }
+
+    return IconChartBar;
+};
+export const SelectTemplate = ({
+    itemsMap,
+    isCustomConfig,
+    setEditorConfig,
+}: {
+    itemsMap: ItemsMap | undefined;
+    isCustomConfig: boolean;
+    isEditorEmpty: boolean;
+    setEditorConfig: (config: string) => void;
+}) => {
+    const loadTemplate = useCallback(
+        (template: TemplateType) => {
+            if (!isCustomConfig) return null;
+
+            /**
+             * When selecting a field for the x axis,
+             * we want to prioritize dimensions and date items
+             * over metrics and table calculations
+             */
+            const sortedItemsForX = Object.values(itemsMap || {}).sort(
+                (a, b) => {
+                    const getPriority = (item: ItemsMap[string]) => {
+                        if (isDimension(item) && isDateItem(item)) return 1;
+                        if (isDimension(item)) return 2;
+                        if (isCustomDimension(item)) return 3;
+                        if (isMetric(item)) return 4;
+                        return 5; // everything else
+                    };
+                    return getPriority(a) - getPriority(b);
+                },
+            );
+
+            /**
+             * When selecting a field for the y axis (and color/size values),
+             * we want to prioritize numeric metrics and table calculations
+             * over dimensions
+             */
+            const sortedItemsForY = Object.values(itemsMap || {}).sort(
+                (a, b) => {
+                    const getPriorityForY = (item: ItemsMap[string]) => {
+                        if (isMetric(item) && isNumericType(item.type))
+                            return 1;
+                        if (isMetric(item)) return 2;
+                        if (isTableCalculation(item)) return 3;
+                        return 4; // everything else
+                    };
+
+                    return getPriorityForY(a) - getPriorityForY(b);
+                },
+            );
+
+            const xField = sortedItemsForX[0];
+            const [yField, extraField] = sortedItemsForY;
+
+            const templateString = generateVegaTemplate(
+                template,
+                xField,
+                yField,
+                extraField,
+            );
+            setEditorConfig(templateString);
+        },
+        [isCustomConfig, itemsMap, setEditorConfig],
+    );
+
+    return (
+        <Menu {...COLLAPSABLE_CARD_POPOVER_PROPS} width={183} closeOnItemClick>
```

Comment:
183 seems very specific. what is the issue?

Reply:
same width as this other menu, 
![image](https://github.com/user-attachments/assets/dbfc06f5-b6b9-4cf1-846d-be4fec2c60af) , but that one is using `    width: max-content;` but that doesn't work on this menu because of the comment below

Reply:
Should I add a comment ? or add 200px to `COLLAPSABLE_CARD_POPOVER_PROPS` ... do you have a better suggestion ?